docs: restructure overview page (#2367)
oauth2-proxy/oauth2-proxy: docs/docs/configuration/sessions.md

Sessions allow a user's authentication to be tracked between multiple HTTP
requests to a service.

> The OAuth2 Proxy uses cookies to track user sessions. Cookies are encrypted and stored in one of the available session storage backends.

Reviewer: ```suggestion
OAuth2 Proxy uses cookies to track user sessions. Cookies are encrypted and stored in one of the available session storage backends.
```

At present the available backends are (as passed to `--session-store-type`):
- [cookie](#cookie-storage) (default)
- [redis](#redis-storage)

### Generating a Cookie Secret

> For encrypting the session data stored inside a `cookie` a `cookie secret` needs to be generated. 
> To generate a strong cookie secret use one of the below commands:

Reviewer: ```suggestion
The session cookies are encrypted using a `cookie secret`. For generating a proper secret use one of the following commands:
```

- [cookie](#cookie-storage) (default)
- [redis](#redis-storage)

### Generating a Cookie Secret

> The cookie containing session data is encrypted by a string called a cookie secret.   

Reviewer: ```suggestion
For encrypting the session data stored inside a `cookie` a `cookie secret` needs to be generated.
```

Author: I'm not sure about this one. The documentation as-is don't clearly state that the session cookie is encrypted. I prefer this info to be explicit rather than implicit

Reviewer: I made a new suggestion for it :)
